x-pack/platform/plugins/shared/alerting/server/rules_client/methods/preview.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { createEsClientWithMeta } from '../../routes/rule/apis/execute/es_client_meta_wrapper';
+import type {
+  AsyncSearchParams,
+  AsyncSearchStrategies,
+  RuleExecutorOptions,
+  RuleExecutorServices,
+} from '../../types';
+import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
+import type { RulesClientContext } from '../types';
+import { DISABLE_FLAPPING_SETTINGS } from '../../types';
+
+export async function preview(
+  context: RulesClientContext,
+  {
+    esClient,
+    requestParams,
+    ruleType,
+    savedObjectsClient,
+    dataViewsService,
+    searchSourceClient,
+    getSpaceId,
+    share,
+    uiSettingsClient,
+  }: any
+) {
+  try {
+    // const [, { data, dataViews, share }] = await core.getStartServices();
+    const esQueries: any[] = [];
+    const asCurrentUserWithMeta = createEsClientWithMeta(esClient, esQueries);
+
+    const services: RuleExecutorServices = {
+      alertsClient: {
+        isTrackedAlert: () => false,
+        getAlertLimitValue: () => 1000,
+        setAlertLimitReached: () => {},
+        getRecoveredAlerts: () => [],
+        setAlertData: () => {},
+        report: () => {
+          return {
+            uuid: uuidv4(),
+          };
+        },
+      },
+      getDataViews: async () => dataViewsService,
+      getMaintenanceWindowIds: async () => [],
+      getSearchSourceClient: async () => searchSourceClient,
+      savedObjectsClient,
+      scopedClusterClient: {
+        asCurrentUser: asCurrentUserWithMeta,
+      },
+      share,
+      shouldStopExecution: () => false,
+      shouldWriteAlerts: () => false,
+      uiSettingsClient,
+      getAsyncSearchClient: <T extends AsyncSearchParams>(strategy: AsyncSearchStrategies) => {
+        throw new Error('Not implemented');
+      },
+    };
+
+    const rule = {
+      id: uuidv4(),
+      ...requestParams,
+      actions: requestParams.actions || [],
```

Comment:
the actions won't do anything if we're just running the executor, so should just pass in [] always.